utilities/infra.py

```diff
@@ -275,26 +275,26 @@ def wait_for_pods_deletion(pods):
 
 
 def get_pod_container_error_status(pod):
-    pod_instance_status = pod.instance.status
-    # Check the containerStatuses and if any containers is in waiting state, return that information:
-
-    for container_status in pod_instance_status.get("containerStatuses", []):
-        waiting_container = container_status.get("state", {}).get("waiting")
-        if waiting_container:
-            return waiting_container["reason"] if waiting_container.get("reason") else waiting_container
+    try:
+        pod_instance_status = pod.instance.status
+        # Check the containerStatuses and if any containers is in waiting state, return that information:
+        for container_status in pod_instance_status.get("containerStatuses", []):
+            waiting_container = container_status.get("state", {}).get("waiting")
+            if waiting_container:
+                return waiting_container["reason"] if waiting_container.get("reason") else waiting_container
+    except NotFoundError:
+        LOGGER.error(f"Pod {pod.name} was deleted")
```

Comment:
nitpick (as a user can send a pod object that never existed:
```suggestion
        LOGGER.error(f"Pod {pod.name} was not found")
```